src/quo2/components/inputs/address_input/view.cljs

```diff
@@ -60,10 +60,14 @@
         value    (reagent/atom "")
         focused? (atom false)]
     (fn [{:keys [scanned-value theme blur? on-change-text on-blur on-focus on-clear on-scan on-detect-ens
-                 ens-regex
-                 valid-ens?]}]
+                 on-detect-address
+                 ens-regex address-regex
+                 valid-ens-or-address?]}]
```

Comment:
Do we need to keep this as an external prop?
Maybe it is better to keep it as an internal state since we do not need to alter it externally (or am I wrong?)

Reply:
In a real world use case we probably need to do some external calls to ENS smart contract to verify the correctness of the ENS, plus maybe other verifications, so better to delegate the business logic to the application layer and let the component just receive this prop.